bilby/gw/likelihood.py

```diff
@@ -1526,76 +1529,56 @@ def log_likelihood_ratio(self):
         d_inner_h = 0.
         optimal_snr_squared = 0.
         complex_matched_filter_snr = 0.
+
         self.parameters.update(self.get_sky_frame_parameters())
+
         waveform_ratio = self.compute_waveform_ratio(self.parameters)
 
         for interferometer in self.interferometers:
-            per_detector_snr = self.calculate_snrs_relative_binning(waveform_ratio[interferometer.name], interferometer)
-
+            per_detector_snr = self.calculate_snrs_relative_binning(
+                waveform_ratio[interferometer.name], interferometer)
             d_inner_h += per_detector_snr.d_inner_h
-            optimal_snr_squared += np.real(
-                per_detector_snr.optimal_snr_squared)
+            optimal_snr_squared += np.real(per_detector_snr.optimal_snr_squared)
             complex_matched_filter_snr += per_detector_snr.complex_matched_filter_snr
 
         log_l = np.real(d_inner_h) - optimal_snr_squared / 2
         return float(log_l.real)
 
-    def find_maximum_likelihood_waveform(self, initial_parameter_guess,
-                                         parameter_bounds, iterations=10,
-                                         likelihood_threshold=1):
-        prev_log_likelihood = -np.inf
+    def find_maximum_likelihood_parameters(self, parameter_bounds,
+                                           iterations=1, atol=1e-3, maxiter=100, **kwargs):
+        if "args" in kwargs:
+            atol = kwargs["atol"]
+        if "maxiter" in kwargs:
+            maxiter = kwargs["maxiter"]
+        parameter_bounds_list = self.get_parameter_list_from_dictionary(parameter_bounds)
+
         for i in range(iterations):
```

Comment:
Shouldn't we just get rid of `iterations` altogether and just to one iteration? If you think it is better keeping it, I would recommend changing the name of the variable to something more self-explanatory.

Reply:
I don't know how significant the improvement of parameters will be with more than one iteration. We can test and change this later?

Reply:
Sure